render/src/main/java/org/codice/imaging/nitf/render/imagerep/Rgb48ImageRepresentationHandler.java

```diff
@@ -20,10 +20,23 @@
 
 import javax.imageio.stream.ImageInputStream;
 
+import org.codice.imaging.nitf.core.image.PixelJustification;
+
 class Rgb48ImageRepresentationHandler extends AbstractRgbImageRepresentationHandler {
+    private final int bitsToDiscard;
+
     Rgb48ImageRepresentationHandler(final Map<Integer, Integer> bandMap,
-            final int actualBitsPerPixelPerBand) {
+            final int actualBitsPerPixelPerBand, final PixelJustification pixelJustification) {
         super(bandMap, actualBitsPerPixelPerBand);
+        if (actualBitsPerPixelPerBand - Byte.SIZE < 0) {
+            this.bitsToDiscard = 0;
+        } else {
+            if (pixelJustification.equals(PixelJustification.RIGHT)) {
+                this.bitsToDiscard = actualBitsPerPixelPerBand - Byte.SIZE;
+            } else {
+                this.bitsToDiscard = Byte.SIZE;
```

Comment:
Just to be clear, this case is really that we are 16 bits per pixel, and we want to be 8 bits per pixel. So to extend this to the NBPP=12 or NBPP=32 cases, we are really discarding NBPP - Byte.SIZE?